resources/assets/coffee/react/changelog-index/main.coffee

```diff
@@ -16,12 +16,37 @@
 #    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
 ###
 
-{a, div, h1, p, span} = ReactDOMFactories
+{button, div, h1, p, span} = ReactDOMFactories
 el = React.createElement
 
+groupChangelogBuilds = (builds) ->
+  _.groupBy builds, (build) ->
+    # Assumes created_at an iso8601 datetime string and removes the time portion.
+    # Example: 2018-07-06T05:43:21+00:00
+    build.created_at.substr(0, 10)
+
+
 class ChangelogIndex.Main extends React.PureComponent
+  constructor: (props) ->
+    super props
+
+    builds = props.data.builds
+    hasMore = builds.length == props.data.search.limit
```

Comment:
if the total range has the same number of builds as limit, the pager still shows but clicking it triggers `ujs_redirect` on the controller and the rest of `showMore` bursts into flames (which also makes `always` not get called)